apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -1199,6 +1202,30 @@ type FIPSConfig struct {
 	CustomFIPSConfig *CustomConfig `json:"customFIPSConfig,omitempty"`
 }
 
+type AgentSidecarInjectionConfig struct {
+	// Enabled enables Sidecar injections.
+	// Default: true
```

Comment:
```suggestion
	// Default: false
```
This is false in helm chart too and should be false here as well. This feature is pretty niche use case and we wouldn't enable it by default.